internal/abci/accumulator.go

```diff
@@ -308,6 +308,12 @@ func (app *Accumulator) DeliverTx(req abci.RequestDeliverTx) (rdt abci.ResponseD
 		return ret
 	}
 
+	r := new(protocol.TxResult)
+	err = json.Unmarshal(deliverRes.Data, r)
```

Comment:
I don't like this. I don't want to have to marshal and unmarshal the TX result, just to get around limitations imposed by reusing Tendermint's response types.